src/lib/operators/abstract_operator.cpp

```diff
@@ -44,16 +43,15 @@ AbstractOperator::~AbstractOperator() {
     bool aborted = transaction_context ? transaction_context->aborted() : false;
     bool left_has_executed = _left_input ? _left_input->executed() : false;
     bool right_has_executed = _right_input ? _right_input->executed() : false;
-    Assert(_executed || aborted || !left_has_executed || !right_has_executed || _consumer_count == 0,
+    Assert(executed() || aborted || !left_has_executed || !right_has_executed || _consumer_count == 0,
            "Operator did not execute, but at least one input operator has.");
   }
 }
 
 OperatorType AbstractOperator::type() const { return _type; }
 
 bool AbstractOperator::executed() const {
-  DebugAssert(_executed || !_output, "Did not expect to see an output for an unexecuted operator.");
-  return _executed;
+  return _state == OperatorState::Executed || _state == OperatorState::Cleared;
```

Comment:
Mmmh, we should rename one of the `executed`. This is clearly ambiguous.

Reply:
Maybe `OperatorState::ResultAvailable`?

Reply:
or do you have any suggestions?

Reply:
I think this is only needed in one place where we still have an open discussion about the semantics. Let's come back to this later.